src/main/kotlin/no/nav/familie/ba/sak/behandling/resultat/YtelsePersonUtils.kt

```diff
@@ -0,0 +1,148 @@
+                resultater.add(YtelsePersonResultat.OPPHØRT)
+            }
+
+            if (erInnvilgetSøknad(ytelsePerson = ytelsePerson, segmenterLagtTil = segmenterLagtTil)) {
+                resultater.add(YtelsePersonResultat.INNVILGET)
+            }
+
+            if (erYtelsenEndretTilbakeITid(ytelsePerson = ytelsePerson,
+                                           andeler = andeler,
+                                           segmenterLagtTil = segmenterLagtTil,
+                                           segmenterFjernet = segmenterFjernet)) {
+                resultater.add(YtelsePersonResultat.ENDRET)
+            }
+
+            // Med "rent opphør" (ikke en fagterm) menes at tidspunkt for opphør er flyttet mot venstre i tidslinjen samtidig som
+            // det ikke er gjort andre endringer (lagt til eller fjernet) som det må tas hensyn til i vedtaket.
+            val periodeStartForRentOpphør: YearMonth? =
+                    if (andeler.isEmpty()) {
+                        // Håndtering av teknisk opphør.
+                        TIDENES_MORGEN.toYearMonth()
+                    } else if (resultater.contains(YtelsePersonResultat.OPPHØRT) &&
+                               segmenterLagtTil.isEmpty && segmenterFjernet.size() > 0) {
+
+                        val innvilgetAndelTom = andeler.maxByOrNull { it.stønadTom }?.stønadTom
+                                                ?: throw Feil("Er ytelsen opphørt skal det være satt tom-dato på alle andeler.")
+
+                        if (segmenterFjernet.any { it.tom.toYearMonth() < innvilgetAndelTom }) {
+                            null
+                        } else {
+                            innvilgetAndelTom.plusMonths(1)
+                        }
+                    } else if (resultater.contains(YtelsePersonResultat.OPPHØRT)) {
+                        andeler.maxByOrNull { it.stønadTom }?.stønadTom?.plusMonths(1)
+                        ?: throw Feil("Er ytelsen opphørt skal det være satt tom-dato på alle andeler.")
+                    } else null
+
+            ytelsePerson.copy(
+                    resultater = resultater.toSet(),
+                    periodeStartForRentOpphør = periodeStartForRentOpphør
+            )
+        }
+    }
+
+    private fun erAvslagPåSøknad(ytelsePerson: YtelsePerson, segmenterLagtTil: LocalDateTimeline<AndelTilkjentYtelse>) =
+            ytelsePerson.erFramstiltKravForINåværendeBehandling && segmenterLagtTil.isEmpty
+
+    private fun erInnvilgetSøknad(ytelsePerson: YtelsePerson, segmenterLagtTil: LocalDateTimeline<AndelTilkjentYtelse>) =
+            ytelsePerson.erFramstiltKravForINåværendeBehandling && !segmenterLagtTil.isEmpty
+
+    private fun erYtelsenOpphørt(andeler: List<AndelTilkjentYtelse>) = andeler.none { it.erLøpende() }
+
+    private fun erYtelsenEndretTilbakeITid(ytelsePerson: YtelsePerson,
+                                           andeler: List<AndelTilkjentYtelse>,
```

Comment:
```suggestion
                                           andeler: List<AndelTilkjentYtelse>,
```
er ikke dette oppdatert tilstand? uansett holder kanskje resterende sjekker :)